protocol/signalfx/log_hecv1_json_test.go

```diff
@@ -0,0 +1,226 @@
+package signalfx
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+	"time"
+	"unsafe"
+
+	"github.com/gorilla/mux"
+	"github.com/mailru/easyjson"
+	"github.com/signalfx/golib/v3/errors"
+	"github.com/signalfx/golib/v3/log"
+	"github.com/signalfx/golib/v3/logsink"
+	"github.com/signalfx/golib/v3/pointer"
+	signalfxformat "github.com/signalfx/ingest-protocols/protocol/signalfx/format"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+// at least 3 numData is expected to test entryNil & timeNil
+func getHecV1Data(numData int, entryNil bool, timeNil bool) ([]byte, error) {
```

Comment:
This method is a bit weird in terms of passing `bool` params and allowing the caller to make sure it doesn't throw `NPE`.  Considering this being used only in this test case, i'm fine.

Reply:
its a test case :)